### PR TITLE
feat(#601): Adds webapp healthckech

### PR DESCRIPTION
**Pull request by lordofthejars**

### CODE REVIEW

File: enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/WebAppHealthCheckEnricher.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        String scheme = getScheme().toUpperCase();
+        String path = getPath();
+
+        return new ProbeBuilder().
+            withNewHttpGet().withNewPort(port).withPath(path).withScheme(scheme).endHttpGet().
+            withInitialDelaySeconds(getInitialDelay(readiness)).build();
+
+    }
+
+    private int getInitialDelay(boolean readness) {
```
Comment (dev-gaur) [inline enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/WebAppHealthCheckEnricher.java]: @lordofthejars You missed the 'i' in readiness . At line 89 and 93. It doesn't matter much.. But would be good to maintain naming consistency.

File: enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/WebAppHealthCheckEnricher.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        final int initialDelay = Configs.asInt(getConfig(Config.initialDelay));
+
+        if (initialDelay < 0) {
+            return readness ? DEFAULT_DELAY_READNESS : DEFAULT_DELAY_LIVENESS;
+        }
+
+        return  initialDelay;
```
Comment (dev-gaur) [inline enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/WebAppHealthCheckEnricher.java]: What if I wanted specify two different Initial Delay values for  Readiness and Liveness probes? Even the default values specified by you for Readiness and Liveness probes are different ( 100 and 1000). So, it should be allowed to specifiy two different Initial Delay values for the two probes.
For eg.,
```
<f8-healthcheck-webapp>
<config>
  <initialdelayReadiness>200</initialdelayReadiness>
  <initialDelayLiveness>400</initialDelayLiveness>
</config>
</f8-healthcheck-webapp>
```

Comment (dev-gaur) [reply enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/WebAppHealthCheckEnricher.java]: If we opt for this, then we can eliminate this getInitialDelay( ) method and change to
```
private Probe getProbe(INT InitialDelay) { ... }
```
and

Call it like
```
protected Probe getReadinessProbe() {
        int initialDelay = Configs.asInt(getConfig(Config.initialDelayReadiness));
        initialDelay = initialDelay < 0 ? DEFAULT_DELAY_READINESS : initialDelay;
        return getProbe(initialDelay);
}
```

Comment (dev-gaur) [reply enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/WebAppHealthCheckEnricher.java]: Correct me if I am wrong here.

Comment (lordofthejars) [reply enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/WebAppHealthCheckEnricher.java]: Yes, I agree with you, there is one enricher that is doing in the same way as I am doing here, but SpringBoot and thorntail are doing as you mention here, so I am changing.

**Pull request merged**